app/uk/gov/hmrc/incometaxsessiondata/repositories/SessionDataRepository.scala

```diff
@@ -41,47 +44,66 @@ class SessionDataRepository @Inject()(
     domainFormat = Session.format,
     indexes = Seq(
       IndexModel(
-        Indexes.ascending("sessionID"),
+        Indexes.ascending("sessionId", "internalId", "mtditid"),
         IndexOptions()
-          .name("sessionIDIndex")
+          .name("compoundIndex")
           .unique(true)
       ),
+      IndexModel(
+        Indexes.ascending("mtditid"),
+        IndexOptions()
+          .name("sessionIDIndex")
+          .unique(false)
+      ),
       IndexModel(
         Indexes.ascending("lastUpdated"),
         IndexOptions()
           .name("lastUpdatedIndex")
           .expireAfter(config.cacheTtl, TimeUnit.SECONDS)
+          .unique(false)
       )
     ),
     replaceIndexes = true
   ) {
 
-  private def dataFilter(sessionId: String): Bson = {
-    import Filters._
-    and(equal("sessionID", sessionId))
+  private def dataFilter(sessionId: String, internalId: String, mtditid: String): Bson = {
+    and(equal("sessionId", sessionId), equal("internalId", internalId), equal("mtditid", mtditid))
+  }
+
+  private def dataFilterMtditid(mtditid: String): Bson = {
+    and(equal("mtditid", mtditid))
   }
 
-  def get(sessionId: String): Future[Option[Session]] = {
+  def get(sessionId: String, internalId: String, mtditid: String): Future[Option[Session]] = {
     collection
-      .find(dataFilter(sessionId))
+      .find(dataFilter(sessionId, internalId, mtditid))
       .headOption()
   }
 
-  def set(data: Session): Future[Boolean] = {
+  def getByMtditid(mtditid: String): Future[Seq[Session]] = {
+    collection
+      .find(dataFilterMtditid(mtditid))
+      .toFuture()
+  }
 
-    val updatedAnswers = data copy (lastUpdated = Instant.now(clock))
+  def set(data: Session): Future[result.UpdateResult] = {
 
-    collection
+    val a = collection
       .replaceOne(
-        filter = dataFilter(data.sessionID),
-        replacement = updatedAnswers,
+        filter = dataFilter(data.sessionId, data.internalId, data.mtditid),
+        replacement = data,
         options = ReplaceOptions().upsert(true)
       )
       .toFuture()
-      .map(_.wasAcknowledged())
+
+
+    Thread.sleep(1000)
+    println("AAAAAAAA" + a)
+    a
   }
 
-  def deleteOne(sessionId: String): Future[Boolean] = {
-    collection.deleteOne(dataFilter(sessionId)).toFuture().map(_.wasAcknowledged())
+  def deleteOne(sessionId: String, internalId: String, mtditid: String): Future[Boolean] = {
```

Comment:
this method is specific to UT/ITs context as I understand, so we need to play a bit with either different design/implementation for the tests where we use delete ; but we don't need this to be available in production code without real need

Reply:
is there a built in mongo method that will clear the contents of the repo? this would allow us to just delete this method